test/jdk/java/awt/image/VolatileImage/AlphaCompositeTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021, JetBrains s.r.o.. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @key headful
+ * @bug 8267116
+ * @summary  AlphaComposite for VolatileImage graphics
+ * @author Alexey Ushakov
+ * @run main AlphaCompositeTest
+ */
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.awt.image.VolatileImage;
+import java.io.File;
+import java.io.IOException;
+
+public class AlphaCompositeTest {
+    public static void main(String[] args) throws IOException {
+        GraphicsConfiguration gc = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration();
+        VolatileImage vi = gc.createCompatibleVolatileImage(100, 100, Transparency.TRANSLUCENT);
+        BufferedImage gold = gc.createCompatibleImage(100, 100, Transparency.TRANSLUCENT);
+        render(gold.createGraphics());
+
+        Graphics2D g2 = vi.createGraphics();
+        do {
+            render(g2);
+        } while (vi.contentsLost());
+
+        BufferedImage snapshot = vi.getSnapshot();
```

Comment:
The snapshot should be saved inside the loop above, overwise content can be lost after the loop but brefore "vi.getSnapshot()"

Reply:
Done